packages/wdio-utils/src/envDetector.ts

```diff
@@ -20,30 +20,31 @@ export function isW3C (capabilities?: Capabilities.DesiredCapabilities) {
         return false
     }
 
-    /**
-     * assume session to be a WebDriver session when
-     * - capabilities are returned
-     *   (https://w3c.github.io/webdriver/#dfn-new-sessions)
-     * - it is an Appium session (since Appium is full W3C compliant)
-     */
-    const isAppium = Boolean(
-        capabilities.automationName ||
-        capabilities.deviceName ||
-        capabilities.appiumVersion
-    )
     const hasW3CCaps = Boolean(
         /**
-         * safari docker image may not provide a platformName therefore
-         * check one of the available "platformName" or "browserVersion"
-         */
-        (capabilities.platformName || capabilities.browserVersion) &&
-        /**
-         * local safari and BrowserStack don't provide platformVersion therefore
-         * check also if setWindowRect is provided
+         * Appium can be JWP (Appium 1.x) or W3C (Appium 1.x and 2.x).
+         * JWP for Appium has `platformName` and `platformVersion` as valid
+         * caps which are W3C valid caps for Desktop.
          */
-        (capabilities.platformVersion || Object.prototype.hasOwnProperty.call(capabilities, 'setWindowRect'))
+        isMobile(capabilities) ?
+            /**
+             * The check `isMobile` is not enough to determine if Appium is running with W3C caps
+             */
+            Object.keys(capabilities).find((cap) => cap.startsWith('appium:')) :
+            (
+            /**
+             * safari docker image may not provide a platformName therefore
+             * check one of the available "platformName" or "browserVersion"
+             */
+                ((capabilities.platformName || capabilities.browserVersion) &&
+            /**
+             * local safari and BrowserStack don't provide platformVersion therefore
+             * check also if setWindowRect is provided
+             */
+            (capabilities.platformVersion || Object.prototype.hasOwnProperty.call(capabilities, 'setWindowRect')))
+            )
     )
-    return Boolean(hasW3CCaps || isAppium)
+    return Boolean(hasW3CCaps)
```

Comment:
Is there an Appium version that doesn't support any W3C? Afaik Appium (at least the last couple of versions) has W3C support. Now this might be different if it is used with a vendor e.g. Sauce Labs but then we explicitly have a check for that.

Reply:
It has W3C support but as mentioned above there is a diff between how it was started (JWP caps or W3C caps) and version `1.x` and `2.x`

Reply:
So what you are saying is that if someone uses Appium e.g. using Sauce Labs then WebdriverIO should pick JWP because of lack of support for W3C? In this case as I mentioned above it would be great to have a `if saucelabs && ...` statement to not generalise this.

Reply:
Hmm, I'm not talking about the lack of support for W3C, I'm saying that `isMobile` is not per definition W3C, not with how it was started and not with the commands it receives. For Appium `2.x` we can say it's W3C, `1.x` is just a "mess" with being able to support both / a mix

Reply:
> For Appium 2.x we can say it's W3C, 1.x is just a "mess" with being able to support both / a mix

What is the intention of this change then. Yes, with Appium 1.x it is kind of ambiguous and currently WebdriverIO assumes W3C support is somewhat given even though there might be issues in some cases.

Reply:
The `isW3C` method is also used in the Sauce Service, but then on the request parameters. By validating it's mobile and W3C it's not Legacy (for Sauce) and thus needs to add the `sauce:options` in the `launcher`, see also https://github.com/webdriverio/webdriverio/blob/f52c8a5851d965255051602ab9d8edf31acf2ba2/packages/wdio-sauce-service/src/utils.ts#L65

Reply:
> The isW3C method is also used in the Sauce Service, but then on the request parameters.

This is a problem. We should modify this part then.

Reply:
Will do, thanks